providers/core.tcpsocket.js

```diff
@@ -102,18 +102,25 @@ Socket_firefox.prototype.resume = function(continuation) {
 };
 
 Socket_firefox.prototype.listen = function(host, port, continuation) {
-  try {
-    this.serverSocket = new ServerSocket(host, port);
-    this.host = host;
-    this.port = port;
-    this.serverSocket.onConnect = this._onConnect.bind(this);
-    this.serverSocket.listen();
-    continuation();
-  } catch (e) {
+  if (typeof this.serverSocket === 'undefined') {
```

Comment:
Yeah, caught that shortly after merging. https://github.com/freedomjs/freedom-for-firefox/commit/47873eddb9d6a18d2d1de43541c6ba60ff146125